base/missing.jl

```diff
@@ -459,7 +459,7 @@ ERROR: `b` is still missing
 macro coalesce(args...)
     expr = :(missing)
     for arg in reverse(args)
-        expr = :((val = $arg) !== missing ? val : $expr)
+        expr = :(!ismissing((val = $arg;)) ? val : $expr)
     end
     return esc(:(let val; $expr; end))
```

Comment:
This seems to have hygiene problems (calls the wrong `ismissing` function). I think this would be correct:
```suggestion
        expr = :(!ismissing((val = $(esc(arg));)) ? val : $expr)
    end
    return :(local val; $expr; end)
```

Reply:
Thanks. I'm very bad at macros programming... Though your suggestion didn't work as-is, I went back to using `let`. Was `local` intended or just a typo?